tensorflow/core/kernels/mkl_concat_op.cc

```diff
@@ -456,56 +545,59 @@ class MklConcatOp : public OpKernel {
+                         ":" + std::to_string(__LINE__);
       OP_REQUIRES_OK(
           context,
           errors::Aborted("Operation received an exception:", error_msg));
     }
   }
 
   void CallEigenVersion(OpKernelContext* context, const OpInputList& values,
-                        const MklDnnShapeList& mkl_input_shapes) {
-    CHECK_EQ(values.size(), mkl_input_shapes.size());
-
-    std::vector<Tensor> converted_values;
+                        const OpInputList& input_mins,
+                        const OpInputList& input_maxes,
+                        const MklDnnShapeList& mkl_input_shapes,
+                        bool quantized_input) {
+    size_t num_mkl_input_shapes = mkl_input_shapes.size();
+    CHECK_EQ(values.size(), num_mkl_input_shapes);
+    std::vector<Tensor> converted_values(num_mkl_input_shapes);
     TensorShapeList tf_input_shapes;
-    for (int i = 0; i < mkl_input_shapes.size(); i++) {
+    for (size_t i = 0; i < num_mkl_input_shapes; ++i) {
       if (mkl_input_shapes[i].IsMklTensor()) {
         // do conversion from MKL to TF
         Tensor tmp_tensor =
             ConvertMklToTF<T>(context, values[i], mkl_input_shapes[i]);
-        converted_values.push_back(tmp_tensor);
+        converted_values[i] = tmp_tensor;
         tf_input_shapes.push_back(mkl_input_shapes[i].GetTfShape());
       } else {
         // no conversion since it is TF tensor already
-        converted_values.push_back(values[i]);
+        converted_values[i] = values[i];
         tf_input_shapes.push_back(values[i].shape());
       }
     }
 
     // Call Eigen concat.
-    eigen_concat_op_.Compute(context, converted_values, tf_input_shapes);
-
-    // Set output Mkl tensor for this op.
-    MklDnnShape dnn_shape_output;
-    dnn_shape_output.SetMklTensor(false);
-    dnn_shape_output.SetDimensions(4);
-    Tensor* output_tensor = nullptr;
-    TensorShape tf_shape_output;
-    tf_shape_output.AddDim(dnn_shape_output.GetSerializeBufferSize());
-    OP_REQUIRES_OK(context,
-                   context->allocate_output(
-                       GetTensorMetaDataIndex(0, context->num_outputs()),
-                       tf_shape_output, &output_tensor));
-    dnn_shape_output.SerializeMklDnnShape(
-        output_tensor->flat<uint8>().data(),
-        output_tensor->flat<uint8>().size() * sizeof(uint8));
+    eigen_concat_op_.Compute(context, converted_values, tf_input_shapes,
+                             input_mins, input_maxes, quantized_input);
+
+    // Get the number of dims from first input since all input tensors
+    // should have same rank.
+    size_t dims = values[0].shape().dims();
+    MklDnnShape output_data_mkl_shape;
+    output_data_mkl_shape.SetMklTensor(false);
+    output_data_mkl_shape.SetDimensions(dims);
+    AllocateOutputSetMklShape(context, 0, output_data_mkl_shape);
+    if (quantized_input) {
+      MklDnnShape output_min_max_mkl_shape;
+      output_min_max_mkl_shape.SetMklTensor(false);
```

Comment:
Do you have to set the shape (dimension) to 1?

Reply:
No need since this is not an MKL tensor and the layout won't be used in other ops.

Reply:
Thank you for your clarification!